include/zephyr/sys/reboot.h

```diff
@@ -34,6 +34,22 @@ extern "C" {
  */
 extern FUNC_NORETURN void sys_reboot(int type);
 
+/**
+ * @brief Obtain reboot type
+ *
+ * Fetch reboot type which caused system startup.
+ * This call extract reboot type vale which was used in previous call to the
+ * sys_rebot() call, if the platform allow to store the value in
+ * the reset retention storage.
+ * Once value is read, the reboot type in the system is cleared to
```

Comment:
Maybe better to introduce call for clear reboot type instead.

Reply:
> Maybe better to introduce call for clear reboot type instead.

Have you considered this from the user perspective? What would fit best with the use cases you are enabling with this API?

Reply:
I will go with separate call for clear the reboot type.